ui/components/app/wallet-overview/aggregated-percentage-overview.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { zeroAddress } from 'ethereumjs-util';
+import {
+  getCurrentCurrency,
+  getSelectedAccount,
+  getShouldHideZeroBalanceTokens,
+  getTokensMarketData,
+} from '../../../selectors';
+
+import { useAccountTotalFiatBalance } from '../../../hooks/useAccountTotalFiatBalance';
+import { isEqualCaseInsensitive } from '../../../../shared/modules/string-utils';
+import { formatValue, isValidAmount } from '../../../../app/scripts/lib/util';
+import { getIntlLocale } from '../../../ducks/locale/locale';
+import {
+  Display,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { Box, Text } from '../../component-library';
+import { getCalculatedTokenAmount1dAgo } from '../../../helpers/utils/util';
+
+// core already has this exported type but its not yet available in this version
+// todo remove this and use core type once available
+type MarketDataDetails = {
+  tokenAddress: string;
+  pricePercentChange1d: number;
+};
+
+export const AggregatedPercentageOverview = () => {
+  const tokensMarketData: Record<string, MarketDataDetails> =
+    useSelector(getTokensMarketData);
+  const locale = useSelector(getIntlLocale);
+  const fiatCurrency = useSelector(getCurrentCurrency);
+  const selectedAccount = useSelector(getSelectedAccount);
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+  // Get total balance (native + tokens)
+  const { totalFiatBalance, orderedTokenList } = useAccountTotalFiatBalance(
+    selectedAccount,
+    shouldHideZeroBalanceTokens,
+  );
+
+  // Memoize the calculation to avoid recalculating unless orderedTokenList or tokensMarketData changes
+  const totalFiat1dAgo = useMemo(() => {
+    return orderedTokenList.reduce((total1dAgo, item) => {
+      if (item.address) {
+        // This is a regular ERC20 token
+        // find the relevant pricePercentChange1d in tokensMarketData
+        // Find the corresponding market data for the token by filtering the values of the tokensMarketData object
+        const found = Object.values(tokensMarketData).find(
+          (data: MarketDataDetails) =>
+            isEqualCaseInsensitive(data.tokenAddress, item.address),
+        );
```

Comment:
The market data in state is in checksum format, so this O(n) search should be avoidable by just doing something like

`tokensMarketData[item.address]` 

or `tokensMarketData[toChecksumAddress(item.address)]`

if we aren't 100% sure the comparator is already checksummed

Reply:
Thnx for pointing that out! will commit this `tokensMarketData[toChecksumAddress(item.address)]` since the orderedList is not checksummed 🙏

Reply:
[81cafa1](https://github.com/MetaMask/metamask-extension/pull/27097/commits/81cafa1d6aa2a1576af5d109f2550ebd713988a6)